pkg/clustermarshaller/clustermarshaller.go

```diff
@@ -28,6 +28,10 @@ func MarshalClusterSpec(clusterSpec *cluster.Spec, datacenterConfig providers.Da
 		marshallables = append(marshallables, clusterSpec.GitOpsConfig.ConvertConfigToConfigGenerateStruct())
 	}
 
+	if clusterSpec.FluxConfig != nil {
```

Comment:
Maybe a add a unit test fir this case?

Reply:
add a unit test which tests that an error is thrown in both are present

Reply:
added a unit test for this specific case as well